supabase/migrations/20240821090741_change_last_status_view_to_parcels_view_to_include_all_events_array.sql

```diff
@@ -0,0 +1,172 @@
+    le.last_event_timestamp,
+    le.last_event_data,
+    le.last_event_workflow_order,
+    ae.all_events
+   FROM ((parcels p
+     LEFT JOIN latest_events le ON (((p.primary_key = le.parcel_id) AND (le.row_num = 1))))
+     LEFT JOIN aggregated_events ae ON ((p.primary_key = ae.parcel_id)))
+  ORDER BY p.primary_key;
+
+
+create or replace view "public"."parcels_plus" as  SELECT parcels.primary_key AS parcel_id,
+    parcels.collection_datetime,
+    parcels.packing_date,
+    parcels.created_at,
+    packing_slots.name AS packing_slot_name,
+    packing_slots."order" AS packing_slot_order,
+    parcels.voucher_number,
+    collection_centres.name AS collection_centre_name,
+    collection_centres.acronym AS collection_centre_acronym,
+    collection_centres.is_delivery,
+    clients.primary_key AS client_id,
+    clients.full_name AS client_full_name,
+    clients.address_postcode AS client_address_postcode,
+    clients.flagged_for_attention AS client_flagged_for_attention,
+    clients.signposting_call_required AS client_signposting_call_required,
+    clients.phone_number AS client_phone_number,
+    clients.is_active AS client_is_active,
+    family_count.family_count,
+    parcels_events.last_event_name AS last_status_event_name,
+    parcels_events.last_event_data AS last_status_event_data,
+    parcels_events.last_event_timestamp AS last_status_timestamp,
+    parcels_events.last_event_workflow_order AS last_status_workflow_order,
+    parcels_events.all_events
+   FROM (((((parcels
+     LEFT JOIN collection_centres ON ((parcels.collection_centre = collection_centres.primary_key)))
+     LEFT JOIN clients ON ((parcels.client_id = clients.primary_key)))
+     LEFT JOIN packing_slots ON ((parcels.packing_slot = packing_slots.primary_key)))
+     LEFT JOIN family_count ON ((family_count.family_id = clients.family_id)))
+     LEFT JOIN parcels_events ON ((parcels_events.parcel_id = parcels.primary_key)))
+  ORDER BY parcels.packing_date DESC;
+
+
+create or replace view "public"."completed_parcels" as  WITH completed_events AS (
+         SELECT events.parcel_id,
+            max(events."timestamp") AS completed_timestamp
+           FROM events
+          WHERE (events.new_parcel_status IN ( SELECT status_order.event_name
+                   FROM status_order
+                  WHERE status_order.is_successfully_completed_event))
+          GROUP BY events.parcel_id
+        )
+ SELECT completed_events.parcel_id,
+    completed_events.completed_timestamp,
+    parcels_plus.family_count,
+    clients.pet_food
+   FROM ((completed_events
+     JOIN parcels_plus ON ((parcels_plus.parcel_id = completed_events.parcel_id)))
+     LEFT JOIN clients ON ((parcels_plus.client_id = clients.primary_key)));
+
+
+create or replace view "public"."reports" as  WITH first_completed_parcel AS (
+         SELECT min(completed_parcels_1.completed_timestamp) AS start_date
+           FROM completed_parcels completed_parcels_1
+        ), list_of_weeks AS (
```

Comment:
Crikey, this had a lot more dependencies than I thought. Making views from other views seems like bad practice, but it's not worth trying to fix here